deep_field_metadetect/gaussmom/gaussmom.py

```diff
@@ -0,0 +1,417 @@
+
+    mask = jnp.where(distance_from_center <= maxrad, 1, 0)
+
+    return mask.astype(int)
+
+
+def get_weighted_moments(obs, T, maxrad=None):
+    """
+    Get weighted moments using this mixture as the weight, including
+    e1,e2,T,s2n etc.  If you just want the raw moments use
+    get_weighted_sums()
+
+    If you want the expected fluxes, you should set the flux to the inverse
+    of the normalization which is 2*pi*sqrt(det)
+
+    Parameters
+    ----------
+    obs: Observation
+        The Observation to compare with. See ngmix.observation.Observation
+        The Observation must have a weight map set
+    maxrad: float, optional
+        If sent, limit moments to within the specified maximum radius
+    with_higher_order: bool, optional
+        If set to True, return higher order moments in the sums/sums_cov
+        arrays.  See ngmix.moments.MOMENTS_NAME_MAP for a map between
+        name and index.
+
+    Returns
+    -------
+    result array with basic sums as well as summary statistics
+    such as e1,e2,T,s2n etc.
+    """
+
+    sums, sums_cov, wsum, _ = get_weighted_sums(
+        obs,
+        T=T,
+        maxrad=maxrad,
+    )
+    return get_weighted_moments_stats(obs, sums, sums_cov, wsum)
+
+
+# @jax.jit
+def get_weighted_sums(gaussmom_obs, T, maxrad):
+    """
+    Compute weighted moment sums and their covariance for a 2D Gaussian model.
+
+    This function evaluates the weighted image moments within a circular region
+    defined by `maxrad`, using a Gaussian weight function with size parameter `T`.
+    It returns the weighted sums, the covariance of the sums, total weight,
+    and the normalization factor for the weight function.
+
+    Parameters
+    ----------
+    gaussmom_obs : object
+        An object containing observation data with the following attributes:
+            - u, v: pixel coordinates
+            - image: observed image data
+            - pixelwise_wgt: inverse variance for each pixel
+            - area: pixel area
+    T : float
+        Size (trace) of the weighting Gaussian used for computing moments.
+    maxrad : float
+        Maximum radius (in u, v coordinates) for including pixels in the calculation.
+
+    Returns
+    -------
+    sums : jax.numpy.ndarray
+        Array of shape (6,) containing the weighted image moment sums:
+            [v, u, u^2 - v^2, 2uv, u^2 + v^2, 1]
+    sums_cov : jax.numpy.ndarray
+        Covariance matrix of shape (6, 6) for the weighted sums.
+    wsum : float
+        Total sum of the weights applied to the image.
+    wt_norm : float
+        Normalization factor for the Gaussian weight function.
+    """
+    vcen = 0
+    ucen = 0
+
+    mompars = [0, 0, T / 2, 0, T / 2]
+
+    vmod = gaussmom_obs.v - vcen
+    umod = gaussmom_obs.u - ucen
+
+    var = 1.0 / (gaussmom_obs.pixelwise_wgt)
+    rad2 = umod * umod + vmod * vmod
+
+    circle_mask = jnp.where(jnp.sqrt(rad2) <= maxrad, 1, 0)
```

Comment:
Is this masked used in ngmix too? I forgot that detail.

Reply:
There is a maxrad variable in ngmix, I am using this mask to mimic that variable